Introduction to shadcn/ui components

sarahhjchung opened the pull request:
**Changes Made**

This PR adds a new page to the Software Engineer Learning Center -> Tech Stacks section. The new page includes:
- An introduction to shadcn/ui components 
- Steps on how to install a shadcn/ui component into your project
- Ways to customize the components to suit your needs

neecofabian commented on Topics/Tech_Stacks/Shadcn.md:
```diff
@@ -0,0 +1,144 @@
+
+#### [Creating and Customizing Components](#creating-and-customizing-components-1)
+
+## Introduction
```
Consider adding an "Advantages" or "Why use shadcn/ui?" section to introduce its benefits and why it's more useful than other frameworks, such as MUI.

jaspermai replied:
+1 !

sarahhjchung replied:
That's a great idea. I think it makes most sense to add a section like that within the Introduction section so I'll do that.

neecofabian commented on Topics/Tech_Stacks/Shadcn.md:
```diff
@@ -0,0 +1,144 @@
+
+This document will go over how to install and integrate shadcn/ui into your project. Then, it will go through an example of building a `Button` component into your project and talk through the details in each step.
+
+## Installation
```
List different package managers that are compatible for installing shadcn components (ie., npc, pnpm, yarn, etc.) to reiterate that Step 1 in the example is flexible.

neecofabian commented on Topics/Tech_Stacks/Shadcn.md:
```diff
@@ -0,0 +1,144 @@
+
+<!-- This [document](https://ui.shadcn.com/docs/components/button) shows the steps you need to take in order to install and use a `Button` component in your project.  -->
+
+Follow these steps to install, use, and customize a `Button` component in your own project:
```
Do you think readers will benefit from seeing an additional example, for a more complex component? Possible additions can be Alert Dialog or Card

sarahhjchung replied:
Although additional examples may be helpful, I think I agree more with Jasper's point. As an introduction page, I think it's better to focus on one clear, simple example that users can follow. To add more resources, I can also link some additional examples so students can learn more about other components on their own!

neecofabian replied:
I see, it makes sense not to give too much information so the reader isn't overwhelmed.

neecofabian reviewed (changes requested):
Very thorough work on introducing shadcn Sarah! I found some possible additions to give readers more background into the library and how to use it for more complex components.

jaspermai commented on Topics/Tech_Stacks/Shadcn.md:
```diff
@@ -0,0 +1,144 @@
+
+   Similar to variants, you can rename the default sizes or add your own.
+
+See the shadcn/ui [documentation](https://ui.shadcn.com/docs/components/accordion) to see all the components that are available to re-use in your project.
```
@neecofabian @sarahhjchung This is an open-ended thought, but what are your thoughts on adding other visual references to this article? Since ShadCN is a visual component library I'm thinking that it'll be relevant. Maybe images of the different buttons as you're progressing through the example? Feel free to discuss!

neecofabian replied:
That's a great idea, I think it can bring some visual differences to this new page and show off the library's components. In addition to showing the button's progress throughout the example, @sarahhjchung can include the different button variants to show how CVA variant properties affect shadcn objects

sarahhjchung replied:
After addressing all the other comments you and Neeco have made, I'll see if there's any good spots to add some visual references in. I think it might be helpful to add screenshots of the different button variants like you're suggesting.

sarahhjchung replied:
After address the comments, I thought it would be best to share a link to view the button variant examples in the shadcn documentation. Personally, it looks cleaner and focuses the attention on the code snippets that I included.

jaspermai replied:
@sarahhjchung that makes sense! Just took a look and I really like the addition of this resources link. As I mentioned in a previous comment, readers who are interested in exploring ShadCN will naturally follow this path to investigating the component library for themselves.

jaspermai reviewed (changes requested):
Overall I really like your contributions! The organization is great. I left some improvements in this PR review to enhance the article and clarify things that the reader may not understand upon first read.

sarahhjchung commented on Topics/Tech_Stacks/Shadcn.md:
```diff
@@ -0,0 +1,144 @@
+
+If you want to manually install shadcn/ui, follow this [tutorial](https://ui.shadcn.com/docs/installation/manual).
+
+## Creating and Customizing Components
```
That's a great point! I'll add "Example" somewhere in the header.

sarahhjchung commented on Topics/Tech_Stacks/Shadcn.md:
```diff
@@ -0,0 +1,144 @@
+
+   You can make changes to the default variants with Tailwind to fit your project's theme.
+
+   You can also rename the variants and add your own variants to create more customization in your component.
```
Sounds good!

neecofabian replied:
I noticed that the example code uses `cva()` to define variants. It may be helpful to introduce CVA more prominently by encouraging readers to learn more about it in: https://cva.style/docs

sarahhjchung replied:
Good idea, thanks for the link!

sarahhjchung replied:
@neecofabian I've included the CVA documentation in a new section called Resources at the bottom of this page!

sarahhjchung commented:
@neecofabian @jaspermai Thanks for all your comments! I've addressed all the feedback and I even added a new section called `Resources` at the bottom of the page. I'd appreciate if you guys could take a look at the new changes when you can

jaspermai commented on Topics/Tech_Stacks/Shadcn.md:
```diff
@@ -0,0 +1,180 @@
+
+You can use shadcn/ui components in any framework that supports React. Typescript is recommended for your projects but Javascript components are also available by making changes to your configurations.
+
+Follow the link below depending on the framework you are using for your project:
```
```suggestion
Follow any of the links below depending on the framework you are using for your project:
```

jaspermai commented on Topics/Tech_Stacks/Shadcn.md:
```diff
@@ -0,0 +1,180 @@
+
+<!-- This [document](https://ui.shadcn.com/docs/components/button) shows the steps you need to take in order to install and use a `Button` component in your project.  -->
+
+Follow these steps to install, use, and customize a `Button` component in your own project:
```
```suggestion
In this example, we will walk through the steps to install, use, and customize a `Button` component in your own project:
```

jaspermai commented on Topics/Tech_Stacks/Shadcn.md:
````diff
@@ -0,0 +1,180 @@
+   }
+   ```
+
+## Resources
````
I really like this section, good job! 🚀

jaspermai reviewed (changes requested):
Just left some very minor grammatical changes that could help the tone of your article

jaspermai reviewed (approved):
Looks good to me, great work Sarah!

neecofabian reviewed (approved):
Awesome job, Sarah. This will be a valuable resource to many readers!
